src/server/game/AuctionHouse/AuctionHouseSearcher.cpp

```diff
@@ -0,0 +1,722 @@
+        SortableAuctionEntriesList::const_iterator itr = auctionEntries.begin();
+        if (searchListRequest.searchInfo.listfrom)
+        {
+            if (searchListRequest.searchInfo.listfrom > auctionEntries.size())
+                itr = auctionEntries.end();
+            else
+                itr += searchListRequest.searchInfo.listfrom;
+        }
+
+        for (; itr != auctionEntries.end(); ++itr)
+        {
+            (*itr)->BuildAuctionInfo(searchResponse->packet);
+
+            if (++count >= MAX_AUCTIONS_PER_PAGE)
+                break;
+        }
+
+        totalCount = auctionEntries.size();
+    }
+    else
+    {
+        // getAll handling
+        for (auto const& pair : searchableAuctionMap)
+        {
+            std::shared_ptr<SearchableAuctionEntry> const& Aentry = pair.second;
+            ++count;
+            Aentry->BuildAuctionInfo(searchResponse->packet);
+
+            if (count >= MAX_GETALL_RETURN)
+                break;
+        }
+
+        totalCount = searchableAuctionMap.size();
+    }
+
+    searchResponse->packet.put<uint32>(0, count);
+    searchResponse->packet << totalCount;
+    searchResponse->packet << uint32(AUCTION_SEARCH_DELAY);
+
+    _responseQueue->Enqueue(searchResponse);
+}
+
+void AuctionHouseWorkerThread::SearchOwnerListRequest(AuctionSearchOwnerListRequest const& searchOwnerListRequest)
+{
+    SearchableAuctionEntriesMap const& searchableAuctionMap = GetSearchableAuctionMap(searchOwnerListRequest.listFaction);
+
+    AuctionSearcherResponse* searchResponse = new AuctionSearcherResponse();
+    searchResponse->playerGuid = searchOwnerListRequest.ownerGuid;
+    searchResponse->packet.Initialize(SMSG_AUCTION_OWNER_LIST_RESULT, (4 + 4 + 4));
+    searchResponse->packet << (uint32)0;                                     // amount place holder
+
+    uint32 count = 0;
+    uint32 totalcount = 0;
+
+    for (auto const& pair : searchableAuctionMap)
+    {
+        if (pair.second->ownerGuid != searchOwnerListRequest.ownerGuid)
+            continue;
+
+        std::shared_ptr<SearchableAuctionEntry> const& auctionEntry = pair.second;
+        auctionEntry->BuildAuctionInfo(searchResponse->packet);
+        ++count;
+        ++totalcount;
+    }
+
+    searchResponse->packet.put<uint32>(0, count);
+    searchResponse->packet << (uint32)totalcount;
+    searchResponse->packet << uint32(AUCTION_SEARCH_DELAY);
+
+    _responseQueue->Enqueue(searchResponse);
+}
+
+void AuctionHouseWorkerThread::SearchBidderListRequest(AuctionSearchBidderListRequest const& searchBidderListRequest)
+{
+    SearchableAuctionEntriesMap const& searchableAuctionMap = GetSearchableAuctionMap(searchBidderListRequest.listFaction);
+
+    AuctionSearcherResponse* searchResponse = new AuctionSearcherResponse();
+    searchResponse->playerGuid = searchBidderListRequest.ownerGuid;
+    searchResponse->packet.Initialize(SMSG_AUCTION_BIDDER_LIST_RESULT, (4 + 4 + 4) + 30000); // pussywizard: ensure there is enough memory
```

Comment:
I know that this change (upfront allocation of 30 012 bytes) doesn't come from your side, but does it make sense to you? I thought the average bidder list shouldn't be that long.
On the other hand, I think upfront allocation can be useful for getAll requests.

Reply:
I did see that but opted to not change it at the time, but yes, it's a bit excessive and ultimately useless. At 148 bytes per auction entry it's effectively preallocating the buffer size for ~202 bidder auction entries which is unlikely to be the case 99% of the time. It's a bit insane to pre-optimize for ultra edge cases especially when the optimization is overall extremely insignificant. Also, we don't do the same for either SMSG_AUCTION_OWNER_LIST_RESULT or SMSG_AUCTION_LIST_RESULT. As such, I have now removed it https://github.com/azerothcore/azerothcore-wotlk/pull/20830/commits/ed83ecd3422511645e2cfdeb93d7cf63d2cab512